### PR TITLE
Fix PA for more accurate campaign finance

### PR DESCRIPTION
**Pull request by justinkenel**
Also alter Finance Loader to use non-view models (since there was no view definition). Add relationships in the Storage definitions, add specialized look up methods.

### CODE REVIEW

**Comment by justinkenel** [comment]
@Dember224 take a look at the finance code for this (as well as the associations, defined on Storage) - should simplify work in other areas as well

File: Finances/load.js [hunk 1]
```diff
@@ -46,34 +89,89 @@ CampaignFinanceLoader.prototype.loadCampaignFinances = async function(finance_ar
...
+      });
+
+      if(candidacy) {
+        console.log('found it!', candidacy);
+        const contributions = finance_object.contributions ?? null;
+        const expenditures = finance_object.expenditures ?? null;
+
+        await this.CampaignFinance.findOrCreate({
+          where: {
+            candidacy_id: candidacy.candidacy_id,
```
Comment (Dember224) [inline Finances/load.js:162]: Just wondering if this handles duplicate finance records OK? For some reason Arizona, for example, will return multiple return multiple responses on the same candidate, with identical information. 

On my first crack at it findOrCreate threw a fit over attempting to create multiple records with the same primary key. 

Making a note (for myself) to test this on AZ to make sure dupes don't break anything. 

Maybe de-duping should be done in the getters for campaign finance? Not sure which is ideal.

Comment (justinkenel) [reply Finances/load.js:162]: findOrCreate will not create if it finds one with the `where` clause, so there shouldn't be an issue. Really, we'll want it to be "Update" to use the new values.

I think that duplicates are probably... fine? from the getters?

File: StateSearches/Pennsylvania/index.js [hunk 2]
```diff
@@ -373,136 +409,273 @@ async function getAggregatedFilerData(params) {
...
 
-        if(!thisYear.length) {
-            // console.log('No records found for this year', x.length, params.year);
-            return null;
-        }
+            starting_amount: cycleStart,
+            contributions: cycleRaised,
+            expenditures: cycleSpent,
+            year: params.year,
+            election_year: params.year,
```
Comment (Dember224) [inline StateSearches/Pennsylvania/index.js:638]: What's the distinction between year and election year?

File: Storage/index.js [hunk 3]
```diff
@@ -208,32 +299,38 @@ async function getModels() {
...
         expenditures: DataTypes.DECIMAL(10,2)
     });
+    Candidacy.hasOne(CampaignFinance, {
+        foreignKey: 'candidacy_id'
+    });
+    CampaignFinance.belongsTo(Candidacy, {
+        foreignKey: 'candidacy_id'
+    });
 
-    const CandidateSearch = sequelize.define('CandidateSearch', {
```
Comment (Dember224) [inline Storage/index.js:212]: Can we put candidate search back for now? The analytics tool I built is based on its existence. Is there any harm in having this model?

**Review by Dember224 — APPROVED** [review]
This looks awesome! I think I'm gonna pull this and uncomment the deletion of candidate search. Just because the analytics tool I've been working on uses it. 

I don't know if there's a drawback to having it but it seems easier to query than doing a whole bunch of eager loading between several models. Plus the view is live on production.

**Comment by justinkenel** [comment]
@Dember224 go for it - can you add a 'create view if not exists' call before it's defined? otherwise it's created as a table on my local machine.

File: Finances/load.js (hunk 1 above)
Comment (Dember224) [reply Finances/load.js:162]: I've just been calling the campaign finance scrapers "getters." Pretty silly but seemed apt.

**Pull request merged**